pyface/data_view/data_models/tests/test_array_data_model.py

```diff
@@ -18,7 +19,12 @@
 from pyface.data_view.value_types.api import (
     FloatValue, IntValue, no_value
 )
-from pyface.data_view.data_models.array_data_model import ArrayDataModel
+# This import results in an error without numpy installed
+# see enthought/pyface#742
+try:
+    from pyface.data_view.data_models.api import ArrayDataModel
+except TraitError:
+    pass
```

Comment:
In this case I think the Look-Before-You-Leap is better, because the `TraitError` is such an overloaded exception and I would want to be sure that we are only excusing for the scenario when numpy is not available.
`requires_numpy` actually checks if `np` (imported from `optional_dependencies`) is None, so doing `if np is not None` before importing `ArrayDataModel` would match the logic of `requires_numpy` and is more explicit about the fact that we are expecting NumPy not being available, not anything else that could cause `TraitError`.
Really `TraitError` should be reserved for validation error but it has been used for so many other purposes, and changing that would break so much code...